src/components/NewClusterModal.js

```diff
@@ -348,136 +344,132 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
         onPrevious: () => this.setState({ viewMode: undefined }),
         contents: h(Fragment, [
-          p({ style: { margin: '0px 0px 14px', lineHeight: '1.5' } }, [
+          p({ style: { marginTop: 0, lineHeight: '1.5' } }, [
             `You are about to create a virtual machine using an unverified Docker image.
             Please make sure that it was created by you or someone you trust, using one of our `,
             h(Link, { href: terraBaseImages, ...Utils.newTabLinkProps }, ['base images.']),
             ' Custom Docker images could potentially cause serious security issues.'
           ]),
           h(Link, { href: safeImageDocumentation, ...Utils.newTabLinkProps }, ['Learn more about creating safe and secure custom Docker images.']),
           p({ style: { lineHeight: '1.5' } }, [
-            'If you\'re confident that your image is safe, click ', b(['CREATE']), ' to use it. Otherwise, click ', b(['BACK']),
-            ' to select another image.'
+            'If you\'re confident that your image is safe, click ', b([!!currentCluster ? 'NEXT' : 'CREATE']),
+            ' to use it. Otherwise, click ', b(['BACK']), ' to select another image.'
           ]),
           div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '1rem' } }, [
-            h(ButtonSecondary,
-              {
-                style: { marginRight: '2rem' }, onClick: () => this.setState({ viewMode: undefined })
-              },
-              ['Back']),
-            h(ButtonPrimary, { onClick: () => this.createCluster() }, ['Create'])
+            h(ButtonSecondary, { style: { marginRight: '2rem' }, onClick: () => this.setState({ viewMode: undefined }) }, ['Back']),
+            h(ButtonPrimary, {
+              onClick: () => !!currentCluster ? this.setState({ viewMode: 'replace' }) : this.createCluster()
+            }, [!!currentCluster ? 'Next' : 'Create'])
           ])
         ])
       })],
-      [isDeleteView, () => ({
-        onPrevious: () => this.setState({ isDeleteView: false }),
+      ['delete', () => ({
+        onPrevious: () => this.setState({ viewMode: undefined }),
         contents: h(Fragment, [
           h(deleteText),
           div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '1rem' } }, [
-            h(ButtonSecondary,
-              { style: { marginRight: '2rem' }, onClick: () => this.setState({ isDeleteView: false }) }, ['CANCEL']),
-            h(ButtonPrimary, {
-              onClick: () => onSuccess(this.deleteCluster())
-            }, ['DELETE'])
+            h(ButtonSecondary, { style: { marginRight: '2rem' }, onClick: () => this.setState({ viewMode: undefined }) }, ['CANCEL']),
+            h(ButtonPrimary, { onClick: () => onSuccess(this.deleteCluster()) }, ['DELETE'])
           ])
         ])
       })],
-      () => ({
+      ['replace', () => ({
+        onPrevious: () => this.setState({ viewMode: isSelectedImageCustom ? 'warning' : undefined }),
```

Comment:
Clicking through this in the double-confirm case, it feels a little surprising to have to click 'back' twice to cancel out completely. Presumably if someone changes their mind, it's because they want to go back and change something on the original form.

Reply:
I thought the behavior was consistent with the notion of "back", which was also consistent with the behavior of the ⬅️. The current behavior feels more like "cancel" to me. However, if we changed the button to "cancel", I'm not sure what to do with the ⬅️.

I don't feel strongly enough to hold up the PR, but strongly enough to share my thoughts while it's being discussed.